solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -120,15 +122,13 @@ public static SolrCore getCore(
   private static void addReplica(String syntheticCollectionName, CoreContainer cores) {
     SolrQueryResponse rsp = new SolrQueryResponse();
     try {
+      CollectionAdminRequest.AddReplica addReplicaRequest =
+          CollectionAdminRequest.addReplicaToShard(syntheticCollectionName, "shard1")
+              .setCreateNodeSet(cores.getZkController().getNodeName());
+      addReplicaRequest.setWaitForFinalState(true);
```

Comment:
Without this, we ended up creating lots of replicas unnecessarily since this would get called repeatedly while replicas were being created